src/pages/index.js

```diff
@@ -76,44 +80,148 @@ const cardSection = new Section(
   ".cards__list"
 );
 
-cardSection.renderItems(initialCards);
+//Instantiation of PopupDelete for deleting Card
+const popupConfirmDelete = new PopupDelete("#delete-modal");
+popupConfirmDelete.setEventListeners();
+
+const popupEditAvatar = new PopupWithForm("#avatar-edit", handleAvatarSubmit);
+popupEditAvatar.setEventListeners();
+
+// cardSection.renderItems(initialCards);
 
 /* ------------------------------------------------------------------------------------------- */
 /*                                   Functions                                                 */
 /* ------------------------------------------------------------------------------------------- */
 
 function createCard(data) {
-  const card = new Card(data, "#card__template", handleImageClick);
+  const card = new Card(
+    data,
+    "#card__template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLikeCard,
+    handleRemoveLikeCard
+  );
+  console.log("Check owner of card just created", card.owner);
+  // console.log("Check created cardId", data._id);
   return card.getView();
 }
 
 function handleImageClick(data) {
   previewCardModal.open(data);
+  // console.log("Check here for id", data.id);
+}
+
+function handleRemoveLikeCard(card) {
+  api.removeLikeCard(card.getId()).then((response) => {
+    console.log("Api response", response);
+    card.likeStatus = response.isLiked;
+    card._setLikeState();
+    // card._handleLikeIcon();
+    // card.removeLikeCard();
+    console.log("Removed Like", card.getId());
+  });
 }
 
+function handleLikeCard(card) {
+  api.likeCard(card.getId()).then((response) => {
+    console.log("Api resposnse", response);
+    card.likeStatus = response.isLiked;
+    // card._handleLikeIcon();
+    // card.likeCard();
+    card._setLikeState();
+    console.log("Liked", card.getId());
+  });
+}
+
+function handleDeleteCard(card) {
+  popupConfirmDelete.open();
+  popupConfirmDelete.setSubmitCallback(() => {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.handlerDeleteCard();
+        console.log("Deleted", card.getId());
+        popupConfirmDelete.close();
+      })
+      .catch((err) => console.error("Error in deleting card", err));
+  });
+}
+
+// function handleLikeCard(card){
+//   api.likeCard(card.getId())
+// }
+
 /* ------------------------------------------------------------------------------------------- */
 /*                                  Event Handlers                                             */
 /* ------------------------------------------------------------------------------------------- */
 
 function handleProfileFormSubmit(inputData) {
-  userInfo.setUserInfo({
-    profileNameData: inputData.title,
-    profileJobData: inputData.description,
-  });
+  profileEditModal.setLoading(true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```